test/acceptance/keyword_arguments_test.rb

```diff
@@ -0,0 +1,120 @@
+require File.expand_path('../acceptance_test_helper', __FILE__)
+
+class KeywordArgumentsTest < Mocha::TestCase
+  include AcceptanceTest
+
+  def setup
+    setup_acceptance_test
+  end
+
+  def teardown
+    teardown_acceptance_test
+  end
+
+  def test_should_match_hash_parameter_with_keyword_args
+    test_result = run_as_test do
+      mock = mock()
+      mock.expects(:method).with(:key => 42)
+      mock.method({ :key => 42 })
+    end
+    assert_passed(test_result)
+    # with strict keyword matching:
+    # assert_failed(test_result)
```

Comment:
I understand why you've done it, but I would prefer not to commit any of these commented-out lines to `main`.